api/handlers.go

```diff
@@ -688,11 +688,8 @@ func postContainersCreate(c *context, w http.ResponseWriter, r *http.Request) {
 
 	container, err := c.cluster.CreateContainer(containerConfig, name, authConfig)
 	if err != nil {
-		if strings.HasPrefix(err.Error(), "Conflict") {
-			httpError(w, err.Error(), http.StatusConflict)
-		} else {
-			httpError(w, err.Error(), http.StatusInternalServerError)
-		}
+		responseStatus := GetHTTPErrorStatusCode(err)
```

Comment:
Note that the example you linked to actually returns a well-defined error for the 404 case; https://github.com/moby/moby/blob/7d62e40f7e4f3c17d229a7687d6fcca5448de813/client/container_create.go#L46-L51

The [`objectNotFoundError`](https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/client/errors.go#L40-L58) can be converted to a HTTP status code with the `GetHTTPErrorStatusCode ` function; https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/api/server/httputils/errors.go#L20-L73

Would it work instead to bump the vendored version of github.com/docker/docker ?

Reply:
@thaJeztah 
quick correction, the API client swarm is using is dated, it's `imageNotFoundError` instead of `objectNotFoundError`
https://github.com/dani-docker/swarm/blob/b44836b5b6cfd3262c5e7eee7208099d6dcdd018/vendor/github.com/docker/docker/client/container_create.go#L48